src/plugins/wizard/public/application/utils/get_top_nav_config.tsx

```diff
@@ -92,7 +97,15 @@ export const getTopNavConfig = (
             return;
           }
           const currentTitle = savedWizardVis.title;
-          savedWizardVis.visualizationState = JSON.stringify(visualizationState);
+          const indexPattern = await indexPatterns.get(visualizationState.indexPattern || '');
+          savedWizardVis.searchSourceFields = {
+            index: indexPattern,
+          };
+          const vizStateWithoutIndex = {
```

Comment:
Nit: I'm okay altering it like this here for now. Ideally though, we should have a function that can take as it's input the raw redux state and give us the saved object as it's result and vice versa. That way we can write specific tests for the transformations and keep it isolated to a single location.

Right now the transformation to saved object and back to root state are happening in two separate locations.

Reply:
Yeah, I agree. I think this is the first change where we explicitly want different data formats for the redux state and the saved object. So we want a function to encapsulate the mapping between the two.

Reply:
Leaving this as is for now since it may take a lil more thought into how to do it well